### DISCUSSION TITLE
Ballの認識が途切れる

### DISCUSSION THREAD

**Discussion by outdachs** [Yowkees/keyball#534] [Q&A] [answered]
組み上げ後は正常に機能してたのですが、2日目にballが反応しなくなりました。
Remapに接続すると右用もBall無しとして認識してる様子でした
トラックボールの基板を外してもう一度組み直すとしばらく使えるのですが
気が付くと認識しなくなってしまいます。
観察していると、認識不良になる直前にOLEDの表示でballの数値が突然マイナス側に振れてます
初の自作キーボードで知識もなく解決できる兆しもありません
どうかお知恵を貸してください、宜しくお願いします
![IMG_1665](https://github.com/Yowkees/keyball/assets/165231043/e2457300-a561-4b86-b0ec-c77628295df5)

**Comment by Ryo624** [comment]
ご連絡ありがとうございます。

ミドル基板の裏側のはんだ付け、トラックボール基板のはんだ付けを確認させていただきたいので、お手数ですが写真を添付していただいてもよろしいでしょうか。

**Reply by outdachs** [reply]
早速のご連絡ありがとうございます
写真を添付しまので宜しくお願いします
![P3280006](https://github.com/Yowkees/keyball/assets/165231043/2829f11b-47ca-4f1a-826c-6f3a8b02e299)
![P3280007](https://github.com/Yowkees/keyball/assets/165231043/5a728f7a-e642-41f5-bbd0-611bd89c203c)
![P3280008](https://github.com/Yowkees/keyball/assets/165231043/70bbbda0-c6f7-423c-9fa7-e4ec18935e54)
![P3280009](https://github.com/Yowkees/keyball/assets/165231043/2af6aa4a-1c71-464c-80e4-c3a699b10e93)

**Comment by Ryo624** [comment]
写真の添付ありがとうございます。

はんだ付けをとても綺麗に実施されていますので、はんだ付けに問題はなさそうに見えます。
ProMicroを入れ替えてもう一度トラックボール基板を奥まで差し込むと接続の状況はどうなりますでしょうか。

**Reply by outdachs** [reply]
左右のProMicroを入換えて試しましたがダメでした
トラックボール基板の差し直しでも変化なく、試しにトラックボール基板のはんだを再加熱して組み直しましたが変化有りません
過去の記載にあったL型7Pコンスルーピンの変形もなく綺麗な状態だと思います。
5回程度組み直すとまれに2分程度通常反応して突然X,Yとも-1／-16／-17の数値（X、Y両方ゾロ目の表示）を行き来するかもしくはまったく反応しなくなります

**Comment by Ryo624** [comment]
ご対応ありがとうございます。

ProMicroの入れ替えや基板のはんだを再加熱しても症状に変わりがないとのことで了解しました。
L字コンスルーも変形が確認されていないとのことですので、おそらくトラックボール基板、センサー側に不具合が出てしまっていると思われます。
白銀ラボのウェブショップからはんだ付け、動作確認済みのものを販売できますので、購入をご希望の場合お問い合わせフォームからご連絡いただけますでしょうか。

**Comment by koron** [comment, marked as answer]
念のためもしもテスターをご所有でしたら、電源を入れた状態でボールがある側のPro MicroのGNDとVCCの電圧を計測してみてもらえないでしょうか?

期待値としては4.5V以上ですが、3V前後に落ちてしまっている場合を懸念しています。

もしもそうなってしまっていたら以下の対策をご検討ください。

* Pro Micro上のジャンパ(SJ1とかJ1とか書かれている)をハンダでショートさせる
* Keyballをパソコン本体に直接か、セルフパワーのハブを通じて接続する
* LEDを消灯する

市販のPro Micro互換機には電圧が低くなっているものが存在します。
(参考資料: [type-C + 3.3V駆動の互換機の存在 / Pro Microとそのバリエーション](https://zenn.dev/koron/articles/9fee38469a8acc#type-c-%2B-3.3v%E9%A7%86%E5%8B%95%E3%81%AE%E4%BA%92%E6%8F%9B%E6%A9%9F%E3%81%AE%E5%AD%98%E5%9C%A8))
低い電圧だと光学センサーを含む、予想外の不具合を生じる可能性があります。
また電圧が下がる要因としては、バスパワーのハブに接続しLEDを点灯させる、というのも考えられます。

**Reply by outdachs** [reply]
koron様、情報ありがとうございます！
実はセンサー故障が疑われると回答頂いて
問い合わせフォームから購入に進んでいたところ
なぜか正常に動き始めました。
リンク先の記事を拝見したのですが、私の基盤にもATmega32U4の表示がありました
また再発するかもしれないという事でしょうか？
まったくの素人でお恥ずかし限りですが、GNDとVCCの電圧というのが
それぞれとどこか別の場所で測定するのか、単にその端子どうしをテスターではかるのか・・・
というレベルの状態です。

それとなぜか今度は左右が不通になってしまい途方にくれてます。
過去スレッド
https://github.com/Yowkees/keyball/discussions/313
から左右のDATAとDATAの通電も確認できたのですが
そもそもPC接続後2，3分は連携できて使える状態なのでどうにも素人判断では解決しそうにありません。
もう少し悪あがきしてみます

**Reply by outdachs** [reply]
koron様
エイヤっと素人の勢いで当該端子にテスターを当てました
ほぼ3ｖの電圧でした
ちなみにPro Microは遊舎工房で買ったもので、リンク先の記事に載ってる物でした

**Reply by outdachs** [reply]
koronさん
Pro Microのジャンパーを施しました
イロイロと頻発した問題がすべてクリアしました
助かりました！

**Reply by koron** [reply]
解決されたようで良かったです。

後の人のため、わかりやすい計測箇所としては
OLEDの基板に表記されている VCC および GND の電圧
になります。

**Reply by outdachs** [reply]
すみません、一時的に作動したので解決したかと思われたのですが
翌日にはまた不具合が発生してしまいました。
トラックボールの挙動は安定していたのですが、
接続後5分もしないうちに
・左右の接続が途切れる
・キーの入力が反応しない
・OLEDの表示がされない、もしくは砂嵐の表示
・LEDが点灯しない
このような症状がランダムに出現します。

素人では解決できる見込みがなかったので
遊舎工房に出向き、技術サポートをお願いしていろいろと調べて貰いました。
トラックボールのセンサーやPro  Micro単体では異常がありませんでした

基板から1番のLEDを撤去し、キースイッチ・OLEDも外し、Rro Microと基板のみの
最小構成で通電してもエラーがでたので基板が原因な可能性が高いという見解でした
ただ時間の制約もあり、原因の特定までは至っておりません。

私は高速で1時間車を走らせたら遊舎工房に足を運ぶことが可能ですが、
私と同じ知識の初心者ユーザーがkeyballの操作性に期待してネット販売で購入し、
不具合でつまづき、悩み、自作キーボードを諦めてしまう
そんな方の参考になればと書き込ませていただきました
また進展がありましたら報告させていただきます。
